core/src/test/java/com/google/googlejavaformat/java/MainTest.java

```diff
@@ -69,4 +74,28 @@ public void cantImportSortRangeYet() throws UsageException {
     assertThat(err.toString()).contains(
         "--sort-imports can currently only apply to the whole file");
   }
+
+  @Test
+  public void preserveOriginalFile() throws Exception {
+    Path from = Paths.get("src/test/resources/com/google/googlejavaformat/java/testdata/");
+    Path temp = testFolder.newFolder("preserve").toPath();
+    try (DirectoryStream<Path> stream = Files.newDirectoryStream(from, "*.output")) {
```

Comment:
I'm not really worried about performance, and testing that the formatter is idempotent is a good idea.